src/OpenTelemetry.AutoInstrumentation.BuildTasks/CheckForInstrumentationPackages.cs

```diff
@@ -0,0 +1,161 @@
+// <copyright file="CheckForInstrumentationPackages.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Microsoft.Build.Framework;
+using NuGet.Versioning;
+
+namespace OpenTelemetry.AutoInstrumentation.BuildTasks;
+
+/// <summary>
+/// Implements a build task that checks if the project references any packages that can be instrumented and if
+/// the respective adapter packages, necessary to instrument the target, are or not part of the project.
+/// </summary>
+public class CheckForInstrumentationPackages : Microsoft.Build.Utilities.Task
+{
+    private const string LogPrefix = "OpenTelemetry.AutoInstrumentation: ";
+
+    private MessageImportance _logMessageImportance = MessageImportance.Low;
+
+    /// <summary>
+    /// Gets or sets the list of instrumentation target items.
+    /// </summary>
+    /// <remarks>
+    /// See the <code>InstrumentationTarget</code>"/> items on the <code>.targets</code> file for a list of the
+    /// required metadata.
+    /// </remarks>
+    [Required]
+    public ITaskItem[]? InstrumentationTargetItems { get; set; }
+
+    /// <summary>
+    /// Gets or sets the list of runtime copy local items.
+    /// </summary>
+    /// <remarks>
+    /// This list generated by the standard build target <code>ResolvePackageAssets</code>.
+    /// </remarks>
+    [Required]
+    public ITaskItem[]? RuntimeCopyLocalItems { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to use verbose logging.
+    /// </summary>
+    public bool UseVerboseLog { get; set; }
+
+    /// <summary>
+    /// Checks if the project references any packages that can be instrumented and if the respective adapter packages
+    /// were already added to the project.
+    /// </summary>
+    /// <returns>True, if the task completed without errors, false otherwise.</returns>
+    public override bool Execute()
+    {
+        _logMessageImportance = UseVerboseLog ? MessageImportance.High : MessageImportance.Low;
+        try
+        {
+            return ExecuteImplementation();
+        }
+        catch (Exception ex)
+        {
+            Log.LogError($"{LogPrefix}{ex}");
```

Comment:
nit: `Log.LogErrorFromException` could be used instead

Reply:
10727e6e08dbb33055b637bf0e29148697d67eec